horiz_interp/horiz_interp_type.F90

```diff
@@ -181,43 +181,44 @@ subroutine stats ( dat, low, high, avg, miss, missing_value, mask )
  end subroutine stats
 
 !######################################################################################################################
+!> @brief Creates a copy of the horiz_interp_type object
  subroutine horiz_interp_type_eq(horiz_interp_out, horiz_interp_in)
-    type(horiz_interp_type), intent(inout) :: horiz_interp_out
-    type(horiz_interp_type), intent(in)    :: horiz_interp_in
+    type(horiz_interp_type), intent(inout) :: horiz_interp_out !< Output object being set
+    type(horiz_interp_type), intent(in)    :: horiz_interp_in !< Input object being copied
 
     if(.not.horiz_interp_in%I_am_initialized) then
       call mpp_error(FATAL,'horiz_interp_type_eq: horiz_interp_type variable on right hand side is unassigned')
     endif
 
-    horiz_interp_out%faci            => horiz_interp_in%faci
-    horiz_interp_out%facj            => horiz_interp_in%facj
-    horiz_interp_out%ilon            => horiz_interp_in%ilon
-    horiz_interp_out%jlat            => horiz_interp_in%jlat
-    horiz_interp_out%area_src        => horiz_interp_in%area_src
-    horiz_interp_out%area_dst        => horiz_interp_in%area_dst
-    horiz_interp_out%wti             => horiz_interp_in%wti
-    horiz_interp_out%wtj             => horiz_interp_in%wtj
-    horiz_interp_out%i_lon           => horiz_interp_in%i_lon
-    horiz_interp_out%j_lat           => horiz_interp_in%j_lat
-    horiz_interp_out%src_dist        => horiz_interp_in%src_dist
+    horiz_interp_out%faci            =  horiz_interp_in%faci
+    horiz_interp_out%facj            =  horiz_interp_in%facj
+    horiz_interp_out%ilon            =  horiz_interp_in%ilon
+    horiz_interp_out%jlat            =  horiz_interp_in%jlat
+    horiz_interp_out%area_src        =  horiz_interp_in%area_src
+    horiz_interp_out%area_dst        =  horiz_interp_in%area_dst
+    horiz_interp_out%wti             =  horiz_interp_in%wti
+    horiz_interp_out%wtj             =  horiz_interp_in%wtj
+    horiz_interp_out%i_lon           =  horiz_interp_in%i_lon
+    horiz_interp_out%j_lat           =  horiz_interp_in%j_lat
+    horiz_interp_out%src_dist        =  horiz_interp_in%src_dist
```

Comment:
This and all the above pointers including one below are not allocated but their assignments are different compared to `found_neighbors' case.

Reply:
Where is that done?

Reply:
They are allocated in other horiz_interp files:

e.g.
https://github.com/laurenchilutti/FMS/blob/42e061f96343dd8e2df0d0338a372dc8c37b61a9/horiz_interp/horiz_interp_conserve.F90#L168-L171

https://github.com/laurenchilutti/FMS/blob/42e061f96343dd8e2df0d0338a372dc8c37b61a9/horiz_interp/horiz_interp_bilinear.F90#L288-L291

https://github.com/laurenchilutti/FMS/blob/42e061f96343dd8e2df0d0338a372dc8c37b61a9/horiz_interp/horiz_interp_spherical.F90#L217-L220

Reply:
But, found_neighbors is used and allocated nowhere in FMS that I can find

Reply:
It is done here:
interface assignment(=)
  module procedure horiz_interp_type_eq
end interface